src/components/o-modal/_template.js

```diff
@@ -0,0 +1,23 @@
+import html from 'nanohtml';
+
+export default ({ open }, childrenFragment) => {
+  const closeModal = () => {
+    const modal = document.getElementById('modal-wrapper');
+    modal.style.display = 'none';
+    modal.classList.add('m-modal--hidden');
+  };
+
+  const closeModalByClickingOutside = (event) => {
+    if (event.target.id === 'modal-wrapper') closeModal();
+  };
+
+  return html`
+    <div id="modal-wrapper" class="m-modal ${open ? '' : 'm-modal--hidden'}" onclick="${closeModalByClickingOutside}">
```

Comment:
article would be better here instead of div. 

Reason: 
`The article element represents a complete, or self-contained, composition in a document, page, application, or site and that is, in principle, independently distributable or reusable, e.g. in syndication. This could be a forum post, a magazine or newspaper article, a blog entry, a user-submitted comment, an interactive widget or gadget, or any other independent item of content.` https://html.spec.whatwg.org/multipage/sections.html#the-article-element

And as discussed, never relay on global window objects. please use add event listener instead of using onclick